core/src/main/java/org/apache/accumulo/core/client/admin/NewTableConfiguration.java

```diff
@@ -146,4 +163,116 @@ public NewTableConfiguration enableSummarization(SummarizerConfiguration... conf
     summarizerProps = tmp;
     return this;
   }
+
+  /**
+   * Configures a table's locality groups prior to initial table creation.
+   *
+   * Allows locality groups to be set prior to table creation. Additional calls to this method prior to table creation will overwrite previous locality group
+   * mappings.
+   *
+   * @param groups
+   *          mapping of locality group names to column families in the locality group
+   *
+   * @since 2.0.0
+   */
+  public void setLocalityGroups(Map<String,Set<Text>> groups) {
```

Comment:
@keith-turner I went back and forth initially and at one point considered having two versions, one returning this and one returning void. I eventually went with void to match the return type of the TableOperations version. I can modify to return 'this' instead though. Would you suggest that the attachIterator methods return 'this' as well?

Reply:
Yeah I think both should return this.  IMO its better to match the norms in this class rather than what tableoperations does.  Also it allows method calls the be chained.

Reply:
Updated the methods to return 'this' and created new test to verify behavior.